src/cmake/CMakeBasics.cmake

```diff
@@ -206,6 +206,12 @@ if(CUDA_FOUND)
      #otherwise we let the warning occur. We have to set this in CMAKE_CUDA_FLAGS
      #as it is passed to the device link step, unlike compile_options
      set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xnvlink=--suppress-stack-size-warning")
+     # RAJA things
+     set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-extended-lambda --expt-relaxed-constexpr")
+     if(DEFINED CUDA_ARCH)
+       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -restrict -arch ${CUDA_ARCH}")
```

Comment:
we might want to use `gen_code` etc instead of `arch` to avoid ptx, but we can ponder that some other time.